thespian/system/transport/TCPTransport.py

```diff
@@ -491,13 +490,13 @@ def _cancel_fd_ops(self, errfileno):
             return
         if self._processIntents(errfileno, closed=True):
             return
-        for I,W in enumerate(self._waitingTransmits):
-            del self._waitingTransmits[I]
+        if self._waitingTransmits:
+            W = self._waitingTransmits.pop(0)
             if self._nextTransmitStepCheck(W, errfileno, closed=True):
                 self._waitingTransmits.append(W)
             return
         closed_openSocks = []
-        for I in getattr(self, '_openSockets', []):
+        for I in getattr(self, '_openSockets', {}):
```

Comment:
This is purely for clarity; _openSockets is usually a dict, not a list, so while reading the code, it was confusing to have the default be an empty list instead of an empty dict.